django/utils/timezone.py

```diff
@@ -49,7 +66,17 @@ def get_default_timezone():
 
     This is the time zone defined by settings.TIME_ZONE.
     """
-    return pytz.timezone(settings.TIME_ZONE)
+    if settings.USE_DEPRECATED_PYTZ:
+        import pytz
+        msg = (
+            "The USE_DEPRECATED_PYTZ setting, and support for pytz "
+            "timezones is deprecated in favor of the stdlib zoneinfo module."
+            " Please update your code to use zoneinfo and remove the "
+            "USE_DEPRECATED_PYTZ setting."
+        )
+        warnings.warn(msg, RemovedInDjango50Warning, stacklevel=3)
```

Comment:
I think it's too late for a warning, we should warn when `USE_DEPRECATED_PYTZ` is overridden by users.